### DISCUSSION TITLE
Camera not cented on start.

### DISCUSSION THREAD

**Discussion by yeyazu** [ramokz/phantom-camera#391] [Question / Help Wanted] [answered]
So i'm making a 2d game. The problem i'm having is that when starting the game the camera won't be centered on the player. The Camera will be where it's "pre-placed" in the scene and then jump to the players position. The reason this happens is because the player isn't spawned in right away when the game starts. So the camera doesn't have a target at first. (im setting the target in a script). Not sure how to fix this.

I fixed the issue by setting the processing of the camera nodes to always.

**Comment by ramokz** [comment]
Not entirely sure what the exact problem is. It sounds like an intended system behavior?

Does the camera not center on the player once the target has been set?

**Reply by yeyazu** [reply]
The problem was that when i changed a scene/started the game, the camera first showed the location of where it was "pre-placed" in the scene for half a second and then jumped to track the player. I tried setting the processing of the camera nodes to always and it seems to have fixed the issue for the most part.

**Reply by ramokz** [reply, marked as answer]
If the `follow target` is set via code upon the scene starting, then that is an expected result. As the camera will be placed in the scene before the game runs and shouldn't move until it has a valid target to follow.

If the timing is different when you change the `Process Mode` to `Always`, then that implies that the scene is not fully unpaused, i.e., it is still doing something in the background, when the target is being set via your code. So you are likely bypassing the scene instantiation flow by doing that. Am not sure if there are any risks involved with doing that.

But, ideally, you would want to assign the target before running the scene — especially if the camera position is very different from where the player is spawned. If the player needs to be instantiated via code during runtime, then I would suggest manually setting the camera to the specified position in the editor, if the position is known in advance, or try setting the `PCam`'s position to match the player's on startup. If none of those options work, then it might be worth trying either a tween from a generic position to the player (using 2 separate `PCam2Ds`), or using a simple UI loading overlay to mask a jump such as a simple fade in.

It's not uncommon for games to have a camera jump from place to place upon a scene instantiation, and often comes down to masking that cut in clever ways. Similar to avoiding geometric pop in artifacts when loading a complex 3D environment — the scene needs a bit of time to process before the player should be allowed to see it, as it would be jarring otherwise.

**Reply by yeyazu** [reply]
Thanks for the detailed response i appreciate it!

I haven't encountered any problems with the processing mode set to Always yet.
I can't set the camera position in the editor in advance because there will be multiple entry points.
You are probably right that using a fade in screen on startup/scene change to mask the camera jump is probably the right way to do it.
